src/python/ensembl/production/hive/ensembl_genome_metadata/ChecksumTransfer.py

```diff
@@ -0,0 +1,124 @@
+                    result[species_id] = {
+                        'assembly_acc': assembly_acc,
+                        'seq_regions': {}
+                    }
+
+                if name not in result[species_id]['seq_regions']:
+                    result[species_id]['seq_regions'][name] = {}
+                if code == 'md5_toplevel':
+                    result[species_id]['seq_regions'][name]['md5'] = value
+                elif code == 'sha512t24u_toplevel':
+                    result[species_id]['seq_regions'][name]['sha512t24u'] = value
+            return result
+
+    def deposit_data_in_meta(self, md_uri, data, md5=1, sha512t24u=1):
+        md = DBConnection(md_uri)
+        with md.session_scope() as session:
+            updates = []
+
+            for species_id, species_data in data.items():
+                assembly_acc = species_data['assembly_acc']
+                seq_regions = species_data['seq_regions']
+
+                # Fetch all relevant AssemblySequence records
+                assembly_seqs = session.query(AssemblySequence).join(Assembly).filter(
+                    Assembly.accession == assembly_acc,
+                    AssemblySequence.name.in_(seq_regions.keys())
+                ).all()
+
+                # Map AssemblySequence records to their names for easy access
+                assembly_seq_dict = {seq.name: seq for seq in assembly_seqs}
+
+                # Check Seq_region exists in metadata database but not in our dictionary
+                for seq in assembly_seqs:
+                    if seq.name not in seq_regions:
+                        raise ValueError(
+                            f"AssemblySequence with name {seq.name} exists in metadata database but is not found in the provided data for assembly {assembly_acc}")
+
+                for seq_name, checksums in seq_regions.items():
+                    assembly_seq = assembly_seq_dict.get(seq_name)
+                    if not assembly_seq:
+                        # Check Seq_region for a particular species not found in metadata database
+                        raise ValueError(
+                            f"AssemblySequence with name {seq_name} not found in metadata database for assembly {assembly_acc}")
+
+                    # Prepare the update data
+                    update_data = {}
+                    if md5 and 'md5' in checksums:
+                        update_data['md5'] = checksums['md5']
+                    if sha512t24u and 'sha512t24u' in checksums:
+                        update_data['sha512t24u'] = checksums['sha512t24u']
+
+                    # Add to updates list
+                    if update_data:
+                        update_data['assembly_sequence_id'] = assembly_seq.assembly_sequence_id
+                        updates.append(update_data)
+
+            # Perform the bulk update
+            if updates:
+                session.bulk_update_mappings(AssemblySequence, updates)
+                session.commit()
```

Comment:
The `with` statement finish **always** with a commit or rollback. I am not sure you need to run the commit:
https://github.com/Ensembl/ensembl-py/blob/4aaafb5a12a658f3bfb08eaae593e639d66811f5/src/python/ensembl/database/dbconnection.py#L185